testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/TestListener.java

```diff
@@ -58,5 +93,62 @@ public void testFinished(Description description) throws Exception {
         } else {
             System.out.println("Test finished but did not start: " + fullTestName);
         }
+
+        String phase = readFile(MainAgent.tmpfile);
+
+        if(phase.startsWith("4")) {
+            String polluter = phase.split(" ")[1];
+            if(polluter.equals(fullTestName)) {
+                StateCapture sc = new StateCapture(fullTestName);
+                System.out.println("MainAgent.targetTestName: " + MainAgent.targetTestName +
+                        " fullTestName: " + fullTestName);
+                System.out.println("phase: " + phase);
+                System.out.println("test listener at after!!!!!!!!! Capturing the states!!!!!!!!!!!!!");
+                sc.capture();
+            }
+        }
+        else if(phase.startsWith("diffFieldAfter ")) {
```

Comment:
It seems what we need is some configuration that tells us whether to capture state as XML or load XML as state.